packages/web/src/js/Collapse.ts

```diff
@@ -59,109 +61,149 @@ class Collapse extends BaseComponent {
     return `${this.NAME}`;
   }
 
-  onResize() {
-    this.state.width = window.innerWidth;
-    this.updateTriggerElementHandler();
-    this.updateCollapsibleElementHandler();
+  hideParentSiblings(trigger: HTMLElement) {
+    if (this.meta.parent) {
+      const parentElement = SelectorEngine.findOne(this.meta.parent);
+      const children = parentElement?.children;
+
+      if (!children) {
+        return;
+      }
+
+      for (const item of children) {
+        const itemTrigger = SelectorEngine.findOne(`[${NAME_DATA_TOGGLE}="${NAME}"]`, item);
+        const instance = Collapse.getInstance(itemTrigger);
+
+        if (instance?.state?.open && itemTrigger !== trigger) {
+          instance?.hide();
+        }
+      }
+    }
   }
 
-  appendNodeToParentHandler() {
-    if (this.state.collapsed) {
+  async appendNodeToParent() {
+    if (this.state.open) {
       const contentEl = this.target?.children[0];
       const innerHtml = contentEl ? contentEl?.innerHTML : this.target?.innerHTML;
       const originalHtml = this.parent?.innerHTML;
       if (this.parent) {
         this.parent.innerHTML = `${originalHtml}${innerHtml}`;
       }
-      setTimeout(() => this.target?.remove(), 100);
+
+      await this.target?.remove();
     }
   }
 
-  adjustCollapsibleElementHeightHandler(collapsed: boolean = this.state.collapsed) {
+  adjustCollapsibleChildrenHeight(zeroHeight?: boolean) {
     if (this.target) {
       const content = this.target.children[0];
       const bounds = content.getBoundingClientRect();
-      this.target.style.height = collapsed ? `${bounds.height}px` : '0';
-      if (this.target) EventHandler.trigger(this.target, collapsed ? EVENT_SHOWN : EVENT_HIDDEN);
+      this.target.style.height = `${zeroHeight ? '0' : bounds.height}px`;
+    }
+  }
+
+  adjustCollapsibleElementHeight(open: boolean = this.state.open) {
+    this.adjustCollapsibleChildrenHeight(!open);
+    if (this.target) {
+      EventHandler.trigger(this.target, open ? EVENT_SHOWN : EVENT_HIDDEN);
     }
   }
 
-  updateTriggerElementHandler(collapsed: boolean = this.state.collapsed) {
-    const triggers = selectorEngine.findAll(`[${NAME_DATA_TARGET}="${this.meta.id}"]`);
+  updateTriggerElement(open: boolean = this.state.open) {
+    const triggers = SelectorEngine.findAll(`[${NAME_DATA_TARGET}="${this.meta.id}"]`);
     const updateElement = (element: Element | HTMLElement) => {
       element.setAttribute(ARIA_CONTROLS_ATTRIBUTE, this.meta.id);
-      element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, String(collapsed));
-      element.classList.toggle(CLASSNAME_EXPANDED, collapsed);
-      if (this.meta.hideOnCollapse && collapsed) {
+      element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, String(open));
+      if (this.meta.hideOnCollapse && open) {
         element.remove();
-        this.appendNodeToParentHandler();
-        this.onDestroy();
+        this.appendNodeToParent().then(() => this.onDestroy());
```

Comment:
I have tried to remove `async/await` and use
```
this.appendNodeToParent();
this.onDestroy();
```
here and it is working just fine on my machine. Could you please try it yourself?

Reply:
working on my machine too